sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -0,0 +1,510 @@
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template <class U, class P> explicit annotated_ptr(annotated_ptr<U, P> const & ConvertFrom);
+----
+|
+Constructs the `annotated_ptr` object from the `ConvertFrom` object if the template parameter types are compatible. For example, if the properties on one of the operands conflict with the others then the compiler should issue a compile time error.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+reference operator*() const;
+----
+|
+Returns a reference wrapper which can be used to read or write to the underlying pointer. Reads/Writes using the reference will retain the annotations.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+reference operator[](std::ptrdiff_t Index) const;
+----
+|
+Returns an `annotated_ref` reference wrapper to the object at offset `Index`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr operator+(unsigned long long Offset) const;
+----
+|
+Returns an `annotated_ptr` that points to the location `Offset` distance away from the underlying pointer.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+std::ptrdiff_t operator-(annotated_ptr FromPtr) const;
+----
+|
+Returns the distance between the underlying pointer and the pointer encapsulated by `FromPtr`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+operator bool() const noexcept;
+----
+|
+Returns `false` if the underlying pointer is null, returns `true` otherwise.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+operator T*() noexcept = delete;
+operator const T*() const noexcept = delete;
+----
+|
+Implicit conversion to a pointer to the underlying type `T` is not supported.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+T* get() noexcept;
+const T* get() const noexcept;
+----
+|
+Returns the underlying raw pointer. The raw pointer will not retain the annotations.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator=(const T*) noexcept;
+----
+|
+Allows assignment from an pointer to type `T`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator=(annotated_ptr const&) noexcept;
+----
+|
+Allows assignment from an `annotated_ptr` with the same parameterization.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator++() noexcept;
+----
+|
+Prefix increment operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr operator++() noexcept;
+----
+|
+Postfix increment operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator--() noexcept;
+----
+|
+Prefix decrement operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr operator--() noexcept;
+----
+|
+Postfix decrement operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template<typename propertyT>
+static constexpr bool has_property();
+----
+| Returns true if the `PropertyListT` contains the property specified by `propertyT`. Returns false if it does not.
```

Comment:
To make this consistent with the wording in the properties spec you might want to say:
```suggestion
| Returns true if `PropertyListT` contains the property identified by the property class `propertyT`. Returns false if it does not.
```

Reply:
Updated both `has_property` and `get_property` description based on the text in the properties spec.